src/node_file.h

```diff
@@ -55,6 +55,35 @@ enum class FsStatFsOffset {
 constexpr size_t kFsStatFsBufferLength =
     static_cast<size_t>(FsStatFsOffset::kFsStatFsFieldsNumber);
 
+// the possible file extensions that should be tested
+// 0-6: when packageConfig.main is defined
+// 7-9: when packageConfig.main is NOT defined,
+//      or when the previous case didn't found the file
+const std::vector<std::string> legacy_main_extensions = {"",
+                                                         ".js",
+                                                         ".json",
+                                                         ".node",
+                                                         "/index.js",
+                                                         "/index.json",
+                                                         "/index.node",
+                                                         ".js",
+                                                         ".json",
+                                                         ".node"};
+// define the final index of the algorithm resolution
+// when packageConfig.main is defined.
+const uint8_t legacy_main_extensions_with_main_end = 7;
```

Comment:
Can you make this private in bindingdata? no need to expose these variables to all files that includes the header file.

Reply:
If I do that, won't I need to put the `LegacyMainResolve` method inside the BindingData to be able to access that private variable?

Also, by doing this, why not put `legacy_main_resolve` and the other constants too?

Reply:
Yes. I think we should.